frontend/src/reducers/taskReducer.js

```diff
@@ -98,11 +98,20 @@ export const task = (state = {
       return { ...state, completed: true, error: action.error }
 
     case UPDATE_TASK_REQUESTED:
-      return { ...state, completed: false }
+      return { ...state, completed: false };
     case UPDATE_TASK_SUCCESS:
-      return { ...state, completed: true }
+      // Assuming the updated task data is provided in the action payload as 'task'
+      const updatedTaskData = action.task;
+      // Find the index of the task to be updated
+      const taskIndex = state.data.findIndex(t => t.id === updatedTaskData.id);
+      // Create a new array for tasks with the updated task data
```

Comment:
I think you're trying to find a task in a list of tasks, but this is a single task reducer, so it should have from the back-end the task updated, and I fixed for now on the back-end.